projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -542,17 +542,23 @@ export class IgxDropDownComponent extends IgxDropDownBaseDirective implements ID
         this.onSelection.emit(args);
 
         if (!args.cancel) {
-            this.selection.set(this.id, new Set([newSelection]));
-            if (!this.virtDir) {
-                if (oldSelection) {
-                    oldSelection.selected = false;
+            if (args.newSelection === null ||
+                (args.newSelection.hasOwnProperty('value') && args.newSelection.hasOwnProperty('index') && this.virtDir) ||
+                (args.newSelection instanceof IgxDropDownItemComponent && !args.newSelection.isHeader)) {
+                this.selection.set(this.id, new Set([args.newSelection]));
+                if (!this.virtDir) {
+                    if (oldSelection) {
+                        oldSelection.selected = false;
+                    }
+                    if (newSelection) {
+                        newSelection.selected = true;
+                    }
                 }
-                if (newSelection) {
-                    newSelection.selected = true;
+                if (event) {
+                    this.toggleDirective.close();
                 }
-            }
-            if (event) {
-                this.toggleDirective.close();
+            } else {
+                throw new Error('Please provide a valid in drop-down item for the selection!');
```

Comment:
`... valid in drop-down item...` looks like a leftover

Reply:
Yup, thanks for pointing it out!